aten/src/ATen/core/type.cpp

```diff
@@ -511,7 +511,9 @@ bool TupleType::isSubtypeOfExt(const TypePtr rhs_, std::ostream* why_not) const
   bool names_match = !rhs->schema() || test_names_match(schema(), rhs->schema());
   // co-variant rules for tuples
   return names_match && compare(*rhs, [&](const TypePtr a, const TypePtr b) {
-    return a->isSubtypeOfExt(b, why_not);
+    auto unshaped_a = unshapedType(a);
```

Comment:
This is not correct. Covariance for named tuples just means that if `A <: A'`, then `Tuple(A) <: Tuple(A')`. This appears to have already been the check that was being performed. Unshaping the type tosses out the type hierarchy for tensors entirely.

Reply:
hmm ok then it seems that the real problem underneath is due to tensor type hierarchy. `Tuple[shapedTensor] <: Tuple[unshapedTensor]` but not vice versa. TensorType are not covariant with each other if only one has shape.